r/tests/testthat/test-dplyr-mutate.R

```diff
@@ -589,3 +588,106 @@ test_that("mutate() and transmute() with namespaced functions", {
     tbl
   )
 })
+
+test_that("Can use across() within mutate()", {
+  compare_dplyr_binding(
+    .input %>%
+      mutate(across(c(dbl, dbl2), round)) %>%
+      collect(),
+    tbl
+  )
+
+  compare_dplyr_binding(
+    .input %>%
+      mutate(across(c(dbl, dbl2), list(exp, sqrt))) %>%
+      collect(),
+    tbl
+  )
+
+  # across() arguments not in default order
+  compare_dplyr_binding(
+    .input %>%
+      mutate(across(.fns = round, c(dbl, dbl2))) %>%
+      collect(),
+    tbl
+  )
+
+  # ARROW-17364: .names argument not yet supported for across()
+  expect_error(
+    tbl %>%
+      arrow_table() %>%
+      mutate(across(c(dbl, dbl2), round, .names = "{.col}.{.fn}")) %>%
+      collect(),
+    regexp = "`.names` argument to `across()` not yet supported in Arrow",
+    fixed = TRUE
+  )
+
+  # ellipses (...) are a deprecated argument
+  expect_error(
+    tbl %>%
+      arrow_table() %>%
+      mutate(across(c(dbl, dbl2), round, digits = -1)) %>%
+      collect(),
+    regexp = "`...` argument to `across()` is deprecated in dplyr and not supported in Arrow",
+    fixed = TRUE
+  )
+
+  # alternative ways of specifying .fns - as a list
+  compare_dplyr_binding(
+    .input %>%
+      mutate(across(1:dbl2, list(round))) %>%
+      collect(),
+    tbl
+  )
+
```

Comment:
Good catches there, have added in a couple of failing tests for those that I will make pass.

Reply:
1. `across(1:dbl2, list("fun1" = round, "fun2" = exp))`
Good catch!  I've now fixed it to properly check for the name from the list and not just the function name.

2. ` across(1:dbl2, list("fun1" = round(this_is_not_cool(something_else)), "fun2" = exp))`
This doesn't work in R either. We get the "not supported in Arrow" error, which then pushes it to R, which then gives the error it would have given anyway. I've added a test in for it, but it's a bit clunky and I'm not sure if we need it. What do you think? (It's the test which has `round(sqrt(dbl))` in it).